helix-core/src/main/java/org/apache/helix/controller/rebalancer/GlobalRebalancer.java

```diff
@@ -49,19 +52,29 @@ public RebalanceFailureReason(RebalanceFailureType type, String reason) {
       _reason = reason;
     }
 
-    public RebalanceFailureType get_type() {
+    public RebalanceFailureType getType() {
       return _type;
     }
 
-    public String get_reason() {
+    public String getReason() {
       return _reason;
     }
   }
 
-  void init(HelixManager manager);
-
-  Map<String, IdealState> computeNewIdealState(final CurrentStateOutput currentStateOutput,
-      T clusterData, Map<String, Resource> resourceMap);
+  /**
+   * Compute the new IdealStates for all the resources input. The IdealStates include both the new
+   * partition assignment (in the listFiles) and the new replica state mapping (in the mapFields).
+   *
+   * @param clusterData        The Cluster status data provider.
+   * @param resourceMap        A map containing all the rebalancing resources.
+   * @param currentStateOutput The present Current State of the cluster.
+   * @return A map containing the computed new IdealStates.
+   */
+  Map<String, IdealState> computeNewIdealStates(T clusterData, Map<String, Resource> resourceMap,
+      final CurrentStateOutput currentStateOutput);
 
+  /**
+   * @return Details of the rebalance failure. Null if the computing is done successfully.
+   */
   RebalanceFailureReason getFailureReason();
```

Comment:
After some thoughts, I feel like we should create some exception class, e.g `RebalanceException`. And the compute method throws the exception. It would make more sense IMO.
```
public class RebalanceException {
   //enum should be inside here
}
```

Reply:
It's a good point, allow me to think about it.